tests/Unit/IO/Test_H5.cpp

```diff
@@ -450,6 +450,64 @@ SPECTRE_TEST_CASE("Unit.IO.H5.DatRead", "[Unit][IO][H5]") {
   }
 }
 
+// Test that we can read scalar, rank-1, rank-2, and rank-3 datasets
+SPECTRE_TEST_CASE("Unit.IO.H5.H5Read", "[Unit][IO][H5]") {
+  const std::string h5_file_name("Unit.IO.H5.H5Read.h5");
```

Comment:
You only test doubles but you wrote generic functions so you should write a generic test that tests all possible types support by HDF5.